pkg/ingester/transfer.go

```diff
@@ -618,20 +618,30 @@ func (i *Ingester) findTargetIngester(ctx context.Context) (*ring.IngesterDesc,
 func unshippedBlocks(dir string) (map[string][]string, error) {
 	userIDs, err := ioutil.ReadDir(dir)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "unable to list the directory containing TSDB blocks")
 	}
 
 	blocks := make(map[string][]string, len(userIDs))
 	for _, user := range userIDs {
 		userID := user.Name()
-		blocks[userID] = []string{} // seed the map with the userID to ensure we xfer the WAL, even if all blocks are shipped
+		userDir := filepath.Join(dir, userID)
+
+		// Ensure the user dir is actually a directory. There may be spurious files
+		// in the storage, especially when using Minio in the local development environment.
+		if stat, err := os.Stat(userDir); err == nil && !stat.IsDir() {
```

Comment:
What if `err != nil`?

Reply:
In my opinion, if err is returned here, it will also be returned later, so I didn't flag it.

Reply:
Correct. This is just to skip non dirs, but if any error arise, the next `ioutil.ReadDir()` will fail as well.